File: src/main/kotlin/com/xs0/gqlktx/schema/intro/GqlIntroEnumValue.java
package com.xs0.gqlktx.schema.intro;

import si.datastat.graphql.schema.api.GraphQLObject;
import si.datastat.graphql.schema.api.Wrapped;

@GraphQLObject("__EnumValue")
public class GqlIntroEnumValue {
    private final String name;

    public GqlIntroEnumValue(String name) {
        this.name = name;
    }

    @Wrapped("T!")
    public String getName() {
        return name;
    }

    public String getDescription() {
        return null; // TODO
    }

    public boolean getIsDeprecated() {
        return false; // TODO
    }

    public String getDeprecationReason() {
        return null;
    }
}

File: src/main/kotlin/com/xs0/gqlktx/schema/intro/GqlIntroEnumValue.kt
package com.xs0.gqlktx.schema.intro

import com.xs0.gqlktx.GqlObject
import kotlin.reflect.KClass

@GqlObject("__EnumValue")
data class GqlIntroEnumValue(
    val name: String,
    val description: String?,
    val isDeprecated: Boolean,
    val deprecationReason: String?,
    internal val rawValue: Enum<*>,
    internal val sourceClass: KClass<*>,
)